cpp/include/cudf/strings/detail/gather.cuh

```diff
@@ -230,7 +232,8 @@ rmm::device_uvector<char> gather_chars(StringIterator strings_begin,
   if (output_count == 0) return rmm::device_uvector<char>(0, stream, mr);
 
   auto chars_data = rmm::device_uvector<char>(chars_bytes, stream, mr);
-  auto d_chars    = chars_data.data();
+  cudf::experimental::prefetch::detail::prefetch("manual_prefetch", chars_data, stream);
```

Comment:
If this is a tag, let’s use a more descriptive name.
```suggestion
  cudf::experimental::prefetch::detail::prefetch("gather_prefetch", chars_data, stream);
```

Reply:
Since a couple of people requested this, I'll go ahead and make the change. Since that means that every algorithm will have its own key, the set of things we'll have to enable will be quite large, but perhaps it will be useful since some algorithms compose other algorithms and we might want to profile turning some pieces on and off.

Reply:
I think this is okay as adding and checking each unique string key to `PrefetchConfig::config_values` might add to unintended overhead.